src/admin.rs

```diff
@@ -168,7 +168,7 @@ where
     res.put_i32(5);
     res.put_u8(b'I');
 
-    write_all_half(stream, res).await
+    write_all_half(stream, &res).await
```

Comment:
I don't think this changes anything. `res` is deallocated at the end of this block anyway, so borrowing it doesn't do anything, you might as well pass ownership to `write_all_half()`. Passing ownership != copying.

Reply:
if the function doesn't need the explicit object then it's better to have it accept the reference so it can stay more flexible

Reply:
Another reason we're using reference here is because we don't want to clone or pass ownership of the buffer to the send operation so we pass a reference instead.